packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts

```diff
@@ -421,8 +421,10 @@ export class AuthService {
     await this.tokenRepository.deleteOne(tokenEntity);
   }
 
-  async resendEmailVerification(user: Web2UserEntity): Promise<void> {
-    if (user.status !== UserStatus.PENDING) {
+  async resendEmailVerification(userId: number): Promise<void> {
+    const user = await this.userRepository.findOneById(userId);
```

Comment:
```suggestion
    const user = (await this.userRepository.findOneById(userId)) as Web2UserEntity;
```

and remove `as string` below for `user.email`